### DISCUSSION TITLE
How to Plot Eye Diagram in Qucs-s Directly or Using External Tools?

### DISCUSSION THREAD

**Discussion by rupokdas147** [ra3xdh/qucs_s#1099] [Q&A] [answered]
Hello Qucs-s community,

I am working on designing a Transimpedance Amplifier (TIA) and would like to analyze its performance using an eye diagram. I am seeking advice on the best way to plot an eye diagram in Qucs-s.

Specifically, I would like to know:

1. Direct Method:
Is there any built-in feature or tool in Qucs-s that allows the direct plotting of an eye diagram? For example:

Can I configure the transient simulation with a periodic bit sequence (e.g., PRBS) and overlay the results within Qucs-s itself?

Are there any scripts, plugins, or extensions that facilitate this within Qucs-s?

2. External Post-Processing:
If Qucs-s does not have a built-in way to create eye diagrams, I assume I will need to:

Collect the transient simulation data (time vs. voltage) for a pseudo-random bit sequence.

Export this data to an external tool (e.g., Python or MATLAB) to post-process and overlay the bit periods manually.

If this is the case, could you suggest the most efficient workflow to export and process Qucs-s transient data? Any tips on how to align bit periods and overlay them effectively would also be helpful.

3. Additional Advice:
Are there any recommended techniques or approaches in Qucs-s to simulate and capture multiple bit transitions accurately for high-speed circuits?

Thank you in advance for your help! Any insights, tips, or examples (including scripts or step-by-step guidance) would be greatly appreciated.

**Comment by ra3xdh** [comment, marked as answer]
Qucs-S doesn't have a built-in eye diagram. You may export the simulation data to CSV and do whatever you want using Python, Octave etc. Click with RMB on the trace (not in the diagram field) and select *Export to CSV* Also you can use data converter GUI at the *Tools->Data files converter*. Select the dataset file then select input *Qucs dataset* and output *CSV*.
